src/client/components/filter-menu/time-filter-menu/fixed-time-tab.tsx

```diff
@@ -92,51 +139,286 @@ export class FixedTimeTab extends React.Component<FixedTimeTabProps, FixedTimeTa
+    return inHoverTimeRange && !selectionSet;
+  }
+
+  renderDays(weeks: Date[][], monthStart: Date): JSX.Element[] {
+    const { essence: { timezone, dataCube }, timekeeper } = this.props;
+    const { start, end } = this.state;
+    const startDay = day.floor(start, timezone);
+    const dayBeforeEnd = end && day.shift(end, timezone, -1);
+    const nextMonthStart = month.shift(monthStart, timezone, 1);
+
+    return weeks.map((daysInWeek: Date[], row: number) => {
+      return <div className="week" key={row}> {daysInWeek.map((dayDate: Date, column: number) => {
+        const isPast = dayDate < monthStart;
+        const isFuture = dayDate >= nextMonthStart;
+        const isBeyondMaxRange = dayDate > dataCube.getMaxTime(timekeeper);
+        const isSelected = startDay <= dayDate && dayDate < end;
+        const isSelectedEdgeStart = datesEqual(dayDate, start);
+        const isSelectedEdgeEnd = datesEqual(dayDate, dayBeforeEnd);
+        const className = classNames("day", "value",
+            {
+              "past": isPast,
+              "future": isFuture,
+              "beyond-max-range": isBeyondMaxRange,
+              "selectable": this.getIsSelectable(dayDate),
+              "selected": isSelected,
+              "selected-edge": isSelectedEdgeStart || isSelectedEdgeEnd
+            });
+
+        return <div
+            className={className}
+            key={column}
+            onClick={this.selectDay.bind(this, dayDate)}
+            onMouseEnter={this.calculateHoverTimeRange.bind(this, dayDate)}
+        >{getDayInMonth(dayDate, timezone)}</div>;
+      })}</div>;
+    });
+  }
+  renderCalendar(startDate: Date): JSX.Element[] {
+    const { essence: { timezone } } = this.props;
+    const weeks: Date[][] = monthToWeeks(startDate, timezone, getLocale());
+    const firstWeek = weeks[0];
+    const lastWeek = weeks[weeks.length - 1];
+    const countPrepend = 7 - firstWeek.length;
+    const countAppend = 7 - lastWeek.length;
+    weeks[0] = prependDays(timezone, firstWeek, countPrepend);
+    weeks[weeks.length - 1] = appendDays(timezone, lastWeek, countAppend);
+    return this.renderDays(weeks, startDate);
+  }
+
+  renderCalendarNav(startDate: Date): JSX.Element {
+    const { essence: { timezone } } = this.props;
+
+    return <div className="calendar-nav">
+      <div
+          className="caret left"
+          onClick={this.goToPreviousMonth}
+      >
+        <SvgIcon svg={require("../../../icons/full-caret-left.svg")}/>
+      </div>
+      {formatYearMonth(startDate, timezone)}
+      <div
+          className="caret right"
+          onClick={this.goToNextMonth}
+      >
+        <SvgIcon svg={require("../../../icons/full-caret-right.svg")}/>
+      </div>
+    </div>;
+  }
+  createDateRangePickers = () => {
+    const { timeRanges, numTimeRanges } = this.state;
+    const { essence: { dataCube, timezone }, timekeeper } = this.props;
+    let pickers = [];
+    for (let i = 0; i < numTimeRanges; i++) {
```

Comment:
Use `map` on `timeRanges`

Reply:
I've looked into map and I've also looked into what I would have to do to have the same functionality for that segment of code with map. To be able to have the index i or whatever I need to set the values for startTime, endTime, and the value used in the splice I would have to call indexOf() three times. Correct me if I'm wrong, wouldnt the latter be more inefficient because you'd be indexing the list 3 extra times, rather than just one. Lambda functions have their uses, but in this case it might not be ideal.

Reply:
```
return timeRanges.map((range, idx) => (<React.Fragment>
  <Button type="secondary" title={"Delete"} onClick={() => this.setState( { timeRanges: timeRanges.remove(idx) )} />
      <DateRangePicker
startTime={timeRanges.get(idx)[0]}
endTime={timeRanges.get(idx)[1]}
maxTime={dataCube.getMaxTime(timekeeper)}
timezone={timezone}
onStartChange={this.onStartChange}
onEndChange={this.onEndChange}
index={i}/>
</React.Fragment>));
```

Should do the trick. From performance standpoint, recreating `onClick` handler every render look worse and calling `getMaxTime` on each iteration.